dsl/metadata.go

```diff
@@ -14,6 +14,22 @@ import (
 // While keys can have any value the following names are handled explicitly by
 // goa when set on attributes.
 //
+// `error:field`: maps the attribute name in custom result type to the error
```

Comment:
I'm thinking `struct:error:name` might be more consistent for the name of the metadata. Maybe this is more clear:

>`struct:error:name` identifies the attribute of a result type used to select the returned error when multiple errors are defined on the same method. The value of the field corresponding to the attribute with the `struct:error:name` metadata is matched against the names of the method errors as defined in the design. This makes it possible to define distinct transport mappings for the various errors (for example to return different HTTP status codes). There must be one and exactly one attribute with the `struct:error:name` metadata defined on result types used to define error results.

Reply:
sounds good